File: solutions/r/hamming/1/hamming.R
# This is a stub function to take two strings
# and calculate the hamming distance
hamming <- function(strand1, strand2) {
	if (nchar(strand1) == nchar(strand2)) {
		h <- 0
		for (i in 1:nchar(strand1)) {
			if (substr(strand1, i, i) != substr(strand2, i, i)) { h <- h + 1}
		}
		return (h)
	} else {
		stop("error")
	}
}

File: solutions/go/hamming/1/hamming.go
package hamming

import "errors"

func Distance(a, b string) (int, error) {
	
	var d int = 0;
	if len(a) == len(b) {
		for i := 0; i<len(a); i++ {
			if a[i:i+1] != b[i:i+1] {
				d++;
			}
		}
		return d, nil
	} else {
		return 0, errors.New("bad arguments")
	}
}
